src/shared/user/user.model.js

```diff
@@ -0,0 +1,100 @@
+import mongoose from 'mongoose';
+import crypto from 'crypto';
+import findOrCreate from 'mongoose-findorcreate';
+
+/**
+ * User Schema
+ */
+
+const UserSchema = new mongoose.Schema({
+    login: {
+        type: String,
+        unique: true,
+        required: true
+    },
+    name: {
+        type: String,
+        required: true
+    },
+    email: {
+        type: String,
+        unique: true,
+        required: true
+    },
+    role: {
+        type: String,
+        default: 'user'
+    },
+    facebookId: {type: Number},
+    salt: {type: String},
+    hashedPassword: {type: String}
+}, {
+    collection: 'users',
+    toObject: {
+        transform: (doc, ret) => {
+            delete ret.salt;
+            delete ret.hashedPassword;
+            delete ret.__v;
+            return ret;
+        }
+    }
+});
+
+UserSchema
+    .virtual('password')
+    .set(function (password) {
+        this._password = password;
+        this.salt = makeSalt();
+        this.hashedPassword = encryptPassword(password, this.salt);
+    })
+    .get(function () {
+        return this._password;
+    });
+
+/**
+ * Make salt for password
+ * private method
+ *
+ * @return {String}
+ *
+ */
+let makeSalt = () => {
```

Comment:
I would prefer to use function declaration instead, and prefer const instead of let